src/AssetViewer.ts

```diff
@@ -0,0 +1,300 @@
+   * For animation purposes. The timestamp of the previous render in milliseconds.
+   */
+  private previousRenderTimestampMs: number = 0;
+
+  /**
+   * For animation purposes. The frame used to cancel the animation.
+   */
+  private cancelAnimationFrame: number = 0;
+
+  /**
+   * The scaling basis used, if the model is scaled.
+   */
+  private scalingBasis: number = 1;
+
+  /**
+   * Whether or not the model should be scaled.
+   */
+  private shouldScaleModel: boolean = false;
+
+  /**
+   * Used to determine if the model is already scaled.
+   */
+  private isScaled: boolean = false;
+
+  /**
+   * Whether or not PBR materials should be used.
+   */
+  private shouldUsePBR: boolean = false;
+
+  /**
+   * Fuel Server Bearer Token used in case the resource is private.
+   */
+  private token: string | undefined;
+
+  /**
+   * The HTML element that holds the scene.
+   */
+  private sceneElement: HTMLElement | undefined;
+
+  /**
+   * Once the Asset Viewer is created, it will setup the scene and start the animation loop.
+   *
+   * @param config The Asset Viewer configuration options.
+   */
+  constructor(config: AssetViewerConfig) {
+    this.elementId = config.elementId ?? 'gz-scene';
+    this.token = config.token;
+
+    this.setupVisualization();
+
+    if (this.scene && config.addModelLighting) {
+      this.scene.addModelLighting();
+    }
+
+    this.shouldScaleModel = !!config.scaleModel;
+    this.shouldUsePBR = !!config.enablePBR;
+
+    this.animate();
+  }
+
+  /**
+   * Destroy the scene.
+   */
+  public destroy() {
+    if (this.cancelAnimationFrame) {
+      cancelAnimationFrame(this.cancelAnimationFrame);
+    }
+
+    this.previousRenderTimestampMs = 0;
+
+    if (this.scene) {
+      this.scene.cleanup();
+    }
+  }
+
+  /**
+   * Resize the scene, according to its container's size.
+   */
+  public resize(): void {
+    if (this.scene && this.sceneElement) {
+      this.scene.setSize(this.sceneElement.clientWidth, this.sceneElement.clientHeight);
+    }
+  }
+
+  /**
+   * Position the camera to start visualizing the asset.
+  */
+  public resetView() {
+    const camera = this.scene?.camera;
+    if (camera) {
+      camera.position.x = this.scalingBasis * 1.1;
+      camera.position.y = -this.scalingBasis * 1.4;
+      camera.position.z = this.scalingBasis * 0.6;
+      camera.rotation.x = 67 * Math.PI / 180;
+      camera.rotation.y = 33 * Math.PI / 180;
+      camera.rotation.z = 12 * Math.PI / 180;
+    }
+  }
+
+  /**
+   * Given all the resource URLs, look for its SDF file and render it.
+   * The obtained Object3D will be added to the Scene.
+   *
+   * @param files All the resource's related URLs.
+   */
+  public renderFromFiles(files: string[]): void {
+    if (!this.scene || !this.sdfParser) {
+      return;
+    }
+
+    this.sdfParser.usingFilesUrls = true;
+    this.sdfParser.enablePBR = this.shouldUsePBR;
+
+    // Look for SDF file.
+    const sdfFile = files.find((file) => file.endsWith(".sdf"));
```

Comment:
This is a bit dangerous. A resource could contain multiple SDF files where only 1 should be rendered. Can you add a todo here:
```
TODO: Use the data within metadata.pbtxt to determine which SDF files to render.
```

Reply:
Thank you! Yes, at first I was going to read `model.sdf` but there are many models and worlds that don't follow this structure.

Can we use `metadata.pbtxt` reliably as a source of truth? Is that file always available or required to upload a model or world? If not, what is a good way to determine what SDF to load?